### PR TITLE
Spring Quarkus 

### PR DESCRIPTION
**Pull request by danielpetisme**
Migration to Quarkus Spring compatibility
* Replace spring dependencies for quarkus-spring when possible
* Replcate the `-oauth2-client` by `quarkus-oidc`
* Remove Spring `WebConfigurer`
* Rename the ROLE `ROLE_USER` to `user`
* Use Security Identity rather than `Princnipal`
* Move Content type negociation from method to class level
* Use pure Quarkus testing (with file security)
* Use `quarkus-config-yaml` + migration to quarkus properties.
UPDATE
* Replace Spring bom by quarkus bom

### CODE REVIEW

**Comment by cescoffier** [comment]
let me know when I can start reviewing.

**Comment by danielpetisme** [comment]
Go for it

File: quarkus-spring/pom.xml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+
+        <quarkus-plugin.version>1.4.1.Final</quarkus-plugin.version>
+        <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
+        <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
+        <quarkus.platform.version>1.4.1.Final</quarkus.platform.version>
+
+        <spring-boot.bom.artifact-id>spring-boot-dependencies</spring-boot.bom.artifact-id>
```
Comment (cescoffier) [inline quarkus-spring/pom.xml]: Didn't we say that having the 2 BOMs was not a good idea?

File: quarkus-spring/pom.xml [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        <spring-boot.bom.group-id>org.springframework.boot</spring-boot.bom.group-id>
+        <spring-boot.bom.version>2.2.6.RELEASE</spring-boot.bom.version>
+
+        <surefire-plugin.version>2.22.1</surefire-plugin.version>
+
+        <testcontainers.version>1.11.1</testcontainers.version>
```
Comment (cescoffier) [inline quarkus-spring/pom.xml]: Imported by Quarkus bom.

File: quarkus-spring/pom.xml [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        <rest-assured.version>4.3.0</rest-assured.version>
```
Comment (cescoffier) [inline quarkus-spring/pom.xml]: Imported by Quarkus bom

File: quarkus-spring/src/main/java/me/escoffier/FrontendResource.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.net.URI;
+
+@RestController
+@RequestMapping("/")
+public class FrontendResource {
+
+    //    @Inject
+//    @ResourcePath("keycloak.js")
+//    Template keycloak;
```
Comment (cescoffier) [inline quarkus-spring/src/main/java/me/escoffier/FrontendResource.java]: Any reason to keep this comment?

**Review by cescoffier — APPROVED** [review]
I made a few comments, but nothing critical.

**Review by cescoffier — APPROVED** [review]

**Pull request merged**